src/Tmds.Ssh/EncryptionAlgorithm.cs

```diff
@@ -3,18 +3,21 @@
 
 using System;
 using System.Collections.Generic;
-using System.Security.Cryptography;
 
 namespace Tmds.Ssh;
 
 sealed class EncryptionAlgorithm
 {
+    private delegate byte[] DecryptDelegate(ReadOnlySpan<byte> key, Span<byte> iv, ReadOnlySpan<byte> data, ReadOnlySpan<byte> tag);
+
     private readonly Func<EncryptionAlgorithm, byte[], byte[], HMacAlgorithm?, byte[], IPacketEncoder> _createPacketEncoder;
     private readonly Func<EncryptionAlgorithm, SequencePool, byte[], byte[], HMacAlgorithm?, byte[], IPacketDecoder> _createPacketDecoder;
+    private readonly DecryptDelegate _decryptData;
```

Comment:
This class is meant as a factory for packet decoders and packet encoders. You need a factory for encryption algorithms. Let's keep them separate by renaming the existing `EncryptionAlgorithms` to `PackagingAlgorithms` and then you can use `EncryptionAlgorithms` for the new factory.

Reply:
Will do, I can't say I was happy with adding more stuff on top of this class as it didn't seem like it fit very well. I'll keep them separate as you've recommended.